kaiban-ai/KaibanJS
# Add YouTube video caption tool and related stories/tests

## Description

This pull request introduces the YouTube Video Caption Tool to the KaibanJS project. This tool extracts video captions from YouTube videos using the YouTube Data API and `youtube-captions-scraper`.

## Changes

- Added the `YouTubeVideoCaption` class in `index.js`.
- Implemented functionality to fetch and process captions for a given YouTube video ID.
- Defined the input schema using Zod for validation.

## Requirements

1. **YouTube API Token**
   - You need an API token to authenticate requests to the YouTube Data API.
   - This token should be passed when creating an instance of the `YouTubeVideoCaption` class.

2. **Input Parameters**
   - The tool requires a single input parameter:
     - `videoId`: A string representing the ID of the YouTube video from which you want to extract captions.

## Example Usage

```jsx
import { YouTubeVideoCaption } from './index.js';

// Create an instance of the tool with the required API token
const youtubeCaptionTool = new YouTubeVideoCaption({ token: 'YOUR_YOUTUBE_API_TOKEN' });

// Define the input parameter
const input = {
  videoId: 'VIDEO_ID_HERE',
};

// Call the tool to extract captions
youtubeCaptionTool._call(input).then((captions) => {
  console.log(captions);
}).catch((error) => {
  console.error(error);
});
```

## Checklist

- [x] Added `YouTubeVideoCaption` class.
- [x] Implemented caption extraction functionality.
- [x] Defined input schema using Zod.
- [x] Added unit tests for the new tool.
- [x] Verified that all tests pass.

## How to Test

1. Clone the repository and navigate to the project directory.
2. Install dependencies: `npm install`
3. Run tests: `npm run test`
4. Verify that the `YouTubeVideoCaption` tool works as expected by following the usage instructions in the updated `README.md` file.

---
